micrometer-core/src/main/java/io/micrometer/core/instrument/binder/commonspool2/CommonsObjectPool2Metrics.java

```diff
@@ -0,0 +1,331 @@
+
+    public CommonsObjectPool2Metrics() {
+        this(emptyList());
+    }
+
+    public CommonsObjectPool2Metrics(Iterable<Tag> tags) {
+        this(getMBeanServer(), tags);
+    }
+
+    public CommonsObjectPool2Metrics(MBeanServer mBeanServer, Iterable<Tag> tags) {
+        this.mBeanServer = mBeanServer;
+        this.tags = tags;
+    }
+
+    private static MBeanServer getMBeanServer() {
+        List<MBeanServer> mBeanServers = MBeanServerFactory.findMBeanServer(null);
+        if (!mBeanServers.isEmpty()) {
+            return mBeanServers.get(0);
+        }
+        return ManagementFactory.getPlatformMBeanServer();
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        for (String type : TYPES) {
+            registerMetricsEventually(
+                    type,
+                    (o, tags) -> {
+                        registerGaugeForObject(registry, o, "NumIdle", tags, "The number of instances currently idle in this pool", "objects");
+
+                        registerGaugeForObject(registry, o, "NumWaiters", tags, "The estimate of the number of threads currently blocked waiting for an object from the pool", "threads");
+
+                        registerFunctionCounterForObject(registry, o, "CreatedCount", tags, "The total number of objects created for this pool over the lifetime of the pool", "objects");
+                        registerFunctionCounterForObject(registry, o, "BorrowedCount", tags, "The total number of objects successfully borrowed from this pool over the lifetime of the pool", "objects");
+                        registerFunctionCounterForObject(registry, o, "ReturnedCount", tags, "The total number of objects returned to this pool over the lifetime of the pool", "objects");
+                        registerFunctionCounterForObject(registry, o, "DestroyedCount", tags, "The total number of objects destroyed by this pool over the lifetime of the pool", "objects");
+                        registerFunctionCounterForObject(
+                                registry, o, "DestroyedByEvictorCount", tags, "The total number of objects destroyed by the evictor associated with this pool over the lifetime of the pool", "objects");
+                        registerFunctionCounterForObject(
+                                registry,
+                                o,
+                                "DestroyedByBorrowValidationCount",
+                                tags,
+                                "The total number of objects destroyed by this pool as a result of failing validation during borrowObject() over the lifetime of the pool",
+                                "objects");
+
+                        registerTimeGaugeForObject(
+                                registry, o, "MaxBorrowWaitTimeMillis", "maxBorrowWaitTime", tags, "The maximum time a thread has waited to borrow objects from the pool");
```

Comment:
Please use dot separated names like `max.borrow.wait.time`, which will be camel-cased by certain `NamingConvention` implementations. By keeping dot-separated names in the core code, we maximize the portability of the metric to many different monitoring systems.

Reply:
@jkschneider thanks for pointing it out, fixed in https://github.com/micrometer-metrics/micrometer/pull/1712/commits/32c5c0d576e2b7f9761bfaaa0d2545dfcc5eb089